modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/sqlquery/SqlQueryEnrichment.scala

```diff
@@ -101,23 +107,25 @@ object SqlQueryEnrichment extends ParseableEnrichment {
  * @param output configuration of output context
  * @param ttl cache TTL in milliseconds
  * @param cache actual mutable LRU cache
- * @param connection initialized DB connection (a mutable single-value cache)
  * @param blocker Allows running blocking enrichments on a dedicated thread pool
  */
-final case class SqlQueryEnrichment[F[_]: Monad: DbExecutor](
+final case class SqlQueryEnrichment[F[_]: Monad: DbExecutor: ResourceF](
   schemaKey: SchemaKey,
   inputs: List[Input],
   db: Rdbms,
   query: SqlQueryEnrichment.Query,
   output: Output,
   ttl: Int,
   cache: SqlCache[F],
-  connection: ConnectionRef[F],
   blocker: BlockerF[F]
 ) extends Enrichment {
   private val enrichmentInfo =
     FailureDetails.EnrichmentInformation(schemaKey, "sql-query").some
 
+  private val logger = LoggerFactory.getLogger(getClass)
+
+  private val dataSource = getDataSource(db)
```

Comment:
Previously `ConnectionRef` was created in `CreateSqlQueryEnrichment`. Is there any reason `dataSource` is created here as `private val` instead of passing as constructor arg as before?

Reply:
What would be the benefit of doing so ?
I tried to simplify the old implementation. `dataSource` is never accessed anywhere else than inside `SqlQueryEnrichment`, and doesn't need to be initialized before.

Reply:
To me `CreateSqlQueryEnrichment` seems like a place where we create/init necessary dependencies. Then, after 'creator' does his job, we can use those injected dependencies in our enrichment class and not think about initialising anything. With `dataSource` created here it can be a bit confusing like `oh, here is one more thing we need to the job` 🙂 But it's really a nit and can remain as it is. 

BTW do we need separate those 2 instances of `CreateSqlQueryEnrichment` for sync and Id? Can't we rely on `F[_]`? It looks like the body of those implementations is the same.

Reply:
Yeah you're right, I'll move it to `CreateSqlEnrichment` :+1: 

> It looks like the body of those implementations is the same.

And again you're right !